internal/provider/cluster_resource.go

```diff
@@ -439,6 +438,23 @@ func (r *clusterResource) Create(
 			}
 		}
 
+		if IsKnown(plan.CockroachVersion) {
+			var planType client.PlanType
+			if clusterSpec.Plan != nil {
+				planType = *clusterSpec.Plan
+			} else if serverless.UsageLimits != nil && serverless.UsageLimits.ProvisionedVirtualCpus != nil {
+				planType = client.PLANTYPE_STANDARD
+			} else {
+				planType = client.PLANTYPE_BASIC
+			}
+
+			if planType == client.PLANTYPE_BASIC {
+				resp.Diagnostics.AddError("Invalid Attribute Combination", "cockroach_version is not supported for BASIC clusters.")
```

Comment:
Also, @kathancox you'll find quite a few errors strings that could use a docs review in this file.  They should all look something like this one.  Just look for `AddError`.  If you want I can tag you on each one.